### DISCUSSION TITLE
php7 vers php8

### DISCUSSION THREAD

**Discussion by christianpizot** [JeromeDevome/GRR#378] [Q&A]
Bonjour,
quand je passe de php 7.4 à php 8 en tant qu'administrateur tout marche bien mais les usagés  une fois qu'ils ont saisie leur identifiant et mot de passe tombe sur une page erreur 500 ! une idée ? 
merci
Christian

Numéro de version GRR fichier :  : 4.3.3 - Release-v4.3.3
Numéro de version GRR BDD :  : 0400004
Système d'exploitation :  : Linux webm178.cluster010.gra.hosting.ovh.net 5.15.41-ovh-vps-grsec-zfs-classid #1 SMP Thu May 19 07:42:04 UTC 2022 x86_64
Version PHP : 7.4.33
Base de données :  : mysql => 5.7.42-log

**Comment by JeromeDevome** [comment]
Bonjour, regardez les logs de votre serveur apache.

Le mar. 9 juil. 2024, 17:57, christianpizot ***@***.***> a
écrit :

**Reply by christianpizot** [reply]
Bonjour
ci joint mon fichier, j'ai fait des essais cette après midi mais je ne suis pas assez bon pour analyser les log
[paroisseoullins.net-09-07-2024.log](https://github.com/user-attachments/files/16146797/paroisseoullins.net-09-07-2024.log)

**Comment by christianpizot** [comment]
Bonjour voici l'erreur
Fatal error: Uncaught TypeError: substr(): Argument #3 ($length) must be of type ?int, string given in /home/paroissew/www/grr/include/functions.inc.php:1877 Stack trace: #0 /home/paroissew/www/grr/include/functions.inc.php(1877): substr('000000000000000...', 0, '') #1 /home/paroissew/www/grr/include/session.inc.php(586): compare_ip_adr('2a02:8428:5fc0:...', 'https://www.par...') #2 /home/paroissew/www/grr/login.php(68): grr_opensession('id', 'mdp') #3 {main} thrown in /home/paroissew/www/grr/include/functions.inc.php on line 1877

**Reply by ynaessens** [reply]
Bonjour,
je pense qu'une mise à jour vers la version 4.3.5 est nécessaire. L'erreur relevée dans le log ne correspond pas.
Si cela ne suffit pas, nous étudierons la cause du problème.
Cordialement,
YN

**Comment by christianpizot** [comment]
Bonjour,
Merci pour votre aide
je viens de faire la mise à jour 
Même problème

Numéro de version GRR fichier :  : 4.3.5 - Release-v4.3.5
Numéro de version GRR BDD :  : 0400004
Système d'exploitation :  : Linux webm091.cluster010.gra.hosting.ovh.net 5.15.41-ovh-vps-grsec-zfs-classid #1 SMP Thu May 19 07:42:04 UTC 2022 x86_64
Version PHP : 7.3.33
Base de données :  : mysql => 5.7.42-log
Time : 1721054618
Timezone : Europe/Paris

en mode développeur voici le message d'erreur 

Fatal error: Uncaught TypeError: substr(): Argument #3 ($length) must be of type ?int, string given in /home/paroissew/www/grr/include/functions.inc.php:1883 Stack trace: #0 /home/paroissew/www/grr/include/functions.inc.php(1883): substr('000000000000000...', 0, '') #1 /home/paroissew/www/grr/include/session.inc.php(586): compare_ip_adr('2a02:8428:5fc0:...', 'https://www.par...') #2 /home/paroissew/www/grr/login.php(68): grr_opensession('pierre', 'nicollet') #3 {main} thrown in /home/paroissew/www/grr/include/functions.inc.php on line 1883

**Comment by christianpizot** [comment]
j'ai aussi ces erreurs mais elle ne sont pas blocante.
Warning: Constant OPTION_IP_ADR already defined in /home/paroissew/www/grr/personnalisation/configperso.inc.php on line 70

Warning: Constant SESSION_NAME already defined in /home/paroissew/www/grr/personnalisation/configperso.inc.php on line 79

**Comment by ynaessens** [comment]
Bonjour,
a priori il s'agit d'une régression de la version 3.5.1 à la 4.3.5...
Ce commit https://github.com/JeromeDevome/GRR/commit/b17e1d87c4f3983d500da5794c70edf719a1a92d devrait résoudre le problème.
Pour ce qui est du warning "already defined", une constante php ne peut pas être redéfinie. Je ne vois pas d'autre solution que de modifier config.inc.php. Ce qui diminue l'intérêt du configperso.inc.php, à moins que quelque chose m'ait échappé.
Merci de nous dire si le commit règle bien votre problème.
Cordialement,
YN

**Reply by christianpizot** [reply]
Bonjour
toujours en erreur
Fatal error: Uncaught TypeError: substr(): Argument #3 ($length) must be of type ?int, string given in /home/paroissew/www/grr/include/functions.inc.php:1896 Stack trace: #0 /home/paroissew/www/grr/include/functions.inc.php(1896): substr('', 0, '') #1 /home/paroissew/www/grr/include/session.inc.php(586): compare_ip_adr('2a02:8428:5fc0:...', 'https://www.par...') #2 /home/paroissew/www/grr/login.php(68): grr_opensession('pierre', 'nicollet') #3 {main} thrown in /home/paroissew/www/grr/include/functions.inc.php on line 1896

Pour le warning "already defined" j'ai suprimé  configperso.inc.php et le problème est réglé. Merci
cordialement
CP

**Reply by ynaessens** [reply]
Bonjour,
pouvez-vous essayer d'installer une version 3.5.1d, afin d'éliminer un doute sur l'origine du problème ?
Cordialement,
YN

**Reply by christianpizot** [reply]
Bonjour,
Je n'ai pas pu faire la mise à jour à cause de la base de données ! Mise à jour de la base de données impossible
Cordialement
CP

**Comment by ynaessens** [comment]
Ah oui, bien sûr, il n'est pas possible de rétrograder.
Mon idée était de vérifier que vos paramètres de filtrage sont compatibles avec la version 3.5.1 afin d'orienter les recherches pour trouver une solution.
Si cela ne vous est pas possible, il faudra poursuivre l'analyse à tâtons.
Cordialement,
YN

**Comment by christianpizot** [comment]
Bonjour
J'ai réussi à régler le problème en modifiant dans la base de données la table grr_setting en passant l'item ip_autorise de https://www.paroiseoullins.net à vide. 
Je n'arrivais pas à la vider dans le menu de configuration.
Cela n'explique pas pourquoi cela fonctionnais dans la version 7.4 !
Cordialement
Christian P

**Reply by ynaessens** [reply]
Bonjour,
merci pour votre retour.
Effectivement, je ne comprends pas pourquoi cela fonctionnait avec php 7.4, puisque le paramètre https://www.paroiseoullins.net n'est pas une adresse IP ni une plage d'adresses.
Ainsi il semble que le code de GRR n'est pas en cause.
Cordialement,
YN
